common/src/main/java/com/microsoft/identity/common/internal/providers/microsoft/microsoftsts/MicrosoftStsOAuth2Configuration.java

```diff
@@ -53,4 +83,29 @@ private URL getEndpoint(URL root, String endpoint) {
 
     }
 
+    private OpenIdProviderConfiguration getOpenIdWellKnownConfigForAuthority() {
+        final URL authority = getAuthorityUrl();
+        return getOpenIdWellKnownConfig(authority.getHost(), authority.getPath());
+    }
+
+    OpenIdProviderConfiguration getOpenIdWellKnownConfig(final String host, final String audience) {
```

Comment:
This looks like something that can't be call from the main thread -- if true, we should probably javadoc

Reply:
Let's also annotate the params as `@Nullable` or `@NonNull` as appropriate

Reply:
Addressed. Yes, this is synchronous so it should not be called from the main thread. I've added the javadoc.